openmdao/drivers/scipy_optimizer.py

```diff
@@ -188,6 +189,7 @@ def _setup_driver(self, problem):
         super(ScipyOptimizeDriver, self)._setup_driver(problem)
         opt = self.options['optimizer']
 
+        self.supports._read_only = False
         self.supports['gradients'] = opt in _gradient_optimizers
         self.supports['inequality_constraints'] = opt in _constraint_optimizers
         self.supports['two_sided_constraints'] = opt in _constraint_optimizers
```

Comment:
Let's set these back to `_read__only =True` when done setting.

Reply:
Fixed